libavcodec/qsvdec.c

```diff
@@ -340,7 +340,6 @@ static void qsv_clear_unused_frames(QSVContext *q)
     while (cur) {
         if (cur->used && !cur->surface.Data.Locked && !cur->queued) {
             cur->used = 0;
-            av_frame_unref(cur->frame);
```

Comment:
Downstream may av_frame_ref the cur->frame and put it to encoder reorder queue. You can not reuse for decode output.

Reply:
You are right. There will be problem even if it passes the test.
I found a very simple way to solve the r2r problem and I have pushed to this branch again. You can review the new one.